src/main/java/User.java

```diff
@@ -0,0 +1,71 @@
+
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+public class User implements UserDetails {
+    private static final long serialVersionUID = 8266525488057072269L;
+    private String username;
+    private String password;
+    private Collection<GrantedAuthority> authorities;
+
+    public User(String username, String password, String roles) {
+        super();
+        this.username = username;
+        this.password = password;
+        setRoles(roles);
+    }
+
+    public void setRoles(final String roles) {
+        authorities = new HashSet<GrantedAuthority>();
+
+
+        for (final String role : roles.split(",")) {
```

Comment:
How it works with incorrect input - several commas or comma at the begninning/end. Add unit tests for that.